tests/Integration/TestClientFactory.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Solarium\Tests\Integration;
+
+use Nyholm\Psr7\Factory\Psr17Factory;
+use Solarium\Client;
+use Http\Adapter\Guzzle6\Client as GuzzlePsrClient;
+use Solarium\Core\Client\Adapter\Psr18Adapter;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+
+final class TestClientFactory
+{
+    public static function createWithPsr18Adapter(array $options = null, EventDispatcherInterface $eventDispatcher = null): Client
```

Comment:
shall we keep this? Or should we just use the default `Curl` adapter for those tests like before?

Reply:
For Solrium 5.2 we should fall back to curl.